Scenes/Enemies/TemplateEnemy/TemplateEnemy.gd

```diff
@@ -44,6 +45,16 @@ func update_path():
 	next_destination_idx = 0
 
 func handle_goal_arrival():
-#	print_debug("reached goal inside enemy")
 	emit_signal("reached_goal", 1)
 	queue_free()
+
+func take_damage(damage):
+	flash_red()
+	health -= damage
+	if (health <= 0):
+		queue_free()
+
+func flash_red():
+	$Polygon2D.material.set_shader_param("turn_red", true)
+	yield(get_tree().create_timer(0.05), "timeout")
+	$Polygon2D.material.set_shader_param("turn_red", false)
```

Comment:
We can probably replace this with an AnimationPlayer instead of the timeout in code, which can also do the resizing very easily. If you didn't use one in your tutorials I'll show it to you, because they are amazingly easy.

Reply:
To be clear, we can merge it like this, but we can probably improve it later with an animationplayer

Reply:
I was thinking about using an animation player but I also figured I would get something done quickly and then explore those later because it seemed like a whole separate thing that I needed to learn! figured we might just wanna do all the animation later on